network-formula/network/wicked/routes.sls

```diff
@@ -0,0 +1,63 @@
+{#-
+Salt state file for managing network routes using Wicked
+Copyright (C) 2023 Georg Pfuetzenreuter <mail+opensuse@georg-pfuetzenreuter.net>
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <https://www.gnu.org/licenses/>.
+-#}
+
+{%- from 'network/wicked/map.jinja' import base, base_backup, routes, script -%}
+
+include:
+  - .common
+  - .service
+
+{%- set file = base ~ '/routes' %}
+{%- if salt['file.file_exists'](file) %}
+network_wicked_routes_backup:
+  file.copy:
+    - names:
+      - {{ base_backup }}/routes:
+        - source: {{ file }}
+{%- endif %}
+
+{%- if routes %}
+{%- set shell_routes = [] %}
+
+network_wicked_routes:
+  file.managed:
+    - name: {{ file }}
+    - contents:
+        - {{ pillar.get('managed_by_salt_formula', '# Managed by the network formula') | yaml_encode }}
+      {%- for route, config in routes.items() %}
+      {%- do shell_routes.append(route ~ '_' ~ config.get('gateway', '')) %}
+        -  {{ route }} {{ config.get('gateway', '-') }} {{ config.get('netmask', '-') }} {{ config.get('interface', '-') }} {{ ' '.join(config.get('options', [])) }}
+      {%- endfor %}
+    - mode: '0640'
+
+network_wicked_routes_reload:
+  cmd.run:
+    - name: {{ script }}routes
+    - stateful
+    {%- if salt['file.file_exists'](script ~ 'routes') -%}:
+      - test_name: {{ script }}routes '{{ ','.join(shell_routes) }}' test
+    {%- else -%}: true
```

Comment:
I get what you do here (resulting in `- stateful: true` - but it would still be more readable to make the `- stateful` line part of the if/else to avoid the (no) linebreak magic.

Reply:
Hm, but then it'd be duplicate `- stateful` lines. :-(

Reply:
I know. Nevertheless, this is one of the rare cases where I ask for a (minor) code duplication ;-) because this minor duplication makes the code easier to read and understand.

Reply:
I wonder, is there any reason the script would not exist here? It seems I already have `file: network_wicked_script` in the requires.

Reply:
Upon testing, it seems to behave with the conditional removed.

Reply:
Ah, I remember now - this is for a situation when the script hasn't yet been installed (for example when running the state for the first time) _but_ a changed routing configuration was provided in the pillar.

Reply:
Ok, changed.

Reply:
much better .-)  (no typo in the smiley, one of my eyes is already closed - time for /dev/bed)

Reply:
Good night, thank you for the review!